dev/src/write-batch.ts

```diff
@@ -567,25 +614,55 @@ export class WriteBatch {
     );
 
     if (explicitTransaction) {
+      assert(!useBatch, 'Cannot use batch mode with transactions');
```

Comment:
You also need to make sure that the block above never creates a transaction.

Reply:
N/A if we separate out atomic logic into separate method